### PR TITLE
test: stop using TRACE_TEST_* for system tests

### LINKED ISSUE #859: System test with wrong permission keeps zombie Node process around
When running `npm run system-test` with insufficient permissions, a Node process keeps running in the background blocking `8080`: 

```sh
$ npm run system-test
[...] PERMISSION_DENIED: Missing or insufficient permissions.
$ npm run system-test
[...]Uncaught Error: listen EADDRINUSE :::8080
```

### PR DESCRIPTION
**Pull request by kjin**
Also, switches to `long-door-651` and fixes #859

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -54,6 +54,9 @@ unit_tests: &unit_tests
         command: npm run check-install
     - run:
         name: Run system tests (skipped for PRs)
+        environment:
+          GCLOUD_PROJECT: node-team-test
```
Comment (JustinBeckwith) [inline .circleci/config.yml]: Could we please switch this to `long-door-651`?

Comment (kjin) [reply .circleci/config.yml]: Sure. Could you give me permissions to the GCP project so I can check in a new key file as well?

Comment (JustinBeckwith) [reply .circleci/config.yml]: done!

**Comment by kjin** [comment]
@JustinBeckwith I switched over to the new project, mind taking another look?

File: system-test/package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "express": "^4.16.3",
     "google-auto-auth": "^0.10.0",
     "got": "^8.0.0",
+    "hard-rejection": "^1.0.0",
```
Comment (JustinBeckwith) [inline system-test/package.json]: If we have `--throw-deprecation` enabled, you shouldn't need this :)

Comment (kjin) [reply system-test/package.json]: We are running system tests with all of Node 6, 8, and 10, so that doesn't seem to handle all cases yet...

Comment (kjin) [reply system-test/package.json]: I'll land this now, let me know if I need to make a follow-up!

**Pull request merged**